### ISSUE TITLE
espree.parse mutates options argument

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/espree#329]
Using espree 3.4.1:

```js
const options = { ecmaFeatures: {} }

require('espree').parse('foo', options);

console.log(options.ecmaFeatures);
```

Expected behavior: `options.ecmaFeatures` should still be an empty object.

Actual behavior: `options.ecmaFeatures` is `{ impliedStrict: false }`

This issue came up when I was looking into freezing the default config object for eslint.
